# HG changeset patch
# User Carlo de Falco <carlo.defalco@gmail.com>
# Date 1239110269 -7200
#      Tue Apr 07 15:17:49 2009 +0200
# Node ID 9dc516d361758078b9da9c7cdf4e88f14820d616
# Parent  25cdd60964428c7110d7a971c73f4adeb27cc5e5
Add a test for saving and reading a matrix with the "-ascii" format

diff --git a/test/ChangeLog b/test/ChangeLog
--- a/test/ChangeLog
+++ b/test/ChangeLog
@@ -1,8 +1,13 @@
+2009-04-07  Carlo de Falco  <kingcrimson@tiscali.it>
+
+	* test_io.m: Add a test for saving and reading a matrix with the
+	"-ascii" format.
+
 2009-03-10  Jason Riedy  <jason@acm.org>
 
 	* test_diag_perm.m: Add tests for permuting sparse matrices and
 	for the correct types from interactions between
 	pseudo-scalars (1x1 matrices).
 
 2009-03-10  Jason Riedy  <jason@acm.org>
 
diff --git a/test/test_io.m b/test/test_io.m
--- a/test/test_io.m
+++ b/test/test_io.m
@@ -215,16 +215,23 @@
 %! assert(!isfield(STR,"scalar_fld") && ...
 %! 	STR.matrix_fld == [1.1,2;3,4] && ...
 %! 	STR.string_fld == "Octave" && ...
 %! 	STR.struct_fld.x == 0 && ...
 %! 	STR.struct_fld.y == 1);
 %!
 %! delete struct.dat;
 %!
+%!test
+%! matrix1 = rand(100, 2);
+%! save -ascii matrix.ascii matrix1
+%! matrix2 = load ("matrix.ascii");
+%! assert (matrix1, matrix2, 1e-9)
+%!
+%! delete matrix.ascii;
 
 %% FIXME Disable this test as it writes to stdout and there is no easy
 %% way to recover output. Need to spawn new octave process and pipe stdout
 %% somewhere to treat this case
 %% test/octave.test/io/puts-1.m
 %!#test
 %! puts ("foo\n");
 
